ONSdigital/eq-author-app
# EAR-1001-create-folder

### What is the context of this PR?

This PR introduces the ability to add a new folder via the Author UI. I have also included some refactoring; changing old class components to function components, making some existing components more reusable, and updating Storybook with new components.

### How to review

Try adding a new folder and see if you can update it's QCode. You should not be able to add a page within the folder yet, but it should initialise with an empty page. You should not be able to delete, move or duplicate the folder and the buttons should be disabled.

- Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
  - ensure it can be opened in Author;
  - then, ensure it can be viewed in Runner by pressing the **view survey** button
- Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
